### PR TITLE
adis16495

### PR DESCRIPTION
**Pull request by nunojsa**
# Description

Add support for the adis16495 class of devices and unit tests.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

This was tested running the defined unit tests plus some manual test to read some read only channels as the temperatures and sensors values (without buffering)

**Test Configuration**:
* Hardware: adis16495-3
* OS: RPI4

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: adi/adis16495.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        return self._get_iio_attr("anglvel_x", "filter_low_pass_3db_frequency", False)
+
+    @anglvel_x_filter_low_pass_3db_frequency.setter
+    def anglvel_x_filter_low_pass_3db_frequency(self, value):
+        """ value returned in HZ """
+        self._set_iio_attr("anglvel_x", "filter_low_pass_3db_frequency", False, value)
+
+    @property
+    def anglvel_x_calibbias(self):
+        return self._get_iio_attr("anglvel_x", "calibbias", False)
```
Comment (tfcollins) [inline adi/adis16495.py:121]: Need docstring

File: adi/adis16495.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+    @anglvel_x_calibbias.setter
+    def anglvel_x_calibbias(self, value):
+        self._set_iio_attr("anglvel_x", "calibbias", False, value)
+
+    @property
+    def anglvel_x_calibscale(self):
+        return self._get_iio_attr("anglvel_x", "calibscale", False)
```
Comment (tfcollins) [inline adi/adis16495.py:129]: Need docstring

File: adi/adis16495.py [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        """ value returned in HZ """
+        return self._get_iio_attr("anglvel_y", "filter_low_pass_3db_frequency", False)
+
+    @anglvel_y_filter_low_pass_3db_frequency.setter
+    def anglvel_y_filter_low_pass_3db_frequency(self, value):
+        self._set_iio_attr("anglvel_y", "filter_low_pass_3db_frequency", False, value)
+
+    @property
+    def anglvel_y_calibbias(self):
+        return self._get_iio_attr("anglvel_y", "calibbias", False)
```
Comment (tfcollins) [inline adi/adis16495.py:153]: Need docstring

File: adi/adis16495.py [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+    @anglvel_y_calibbias.setter
+    def anglvel_y_calibbias(self, value):
+        self._set_iio_attr("anglvel_y", "calibbias", False, value)
+
+    @property
+    def anglvel_y_calibscale(self):
+        return self._get_iio_attr("anglvel_y", "calibscale", False)
```
Comment (tfcollins) [inline adi/adis16495.py:161]: Need docstring

File: adi/adis16495.py [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+        """ value returned in HZ """
+        return self._get_iio_attr("anglvel_z", "filter_low_pass_3db_frequency", False)
+
+    @anglvel_z_filter_low_pass_3db_frequency.setter
+    def anglvel_z_filter_low_pass_3db_frequency(self, value):
+        self._set_iio_attr("anglvel_z", "filter_low_pass_3db_frequency", False, value)
+
+    @property
+    def anglvel_z_calibbias(self):
+        return self._get_iio_attr("anglvel_z", "calibbias", False)
```
Comment (tfcollins) [inline adi/adis16495.py:185]: Need docstring

File: adi/adis16495.py [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+
+    @anglvel_z_calibbias.setter
+    def anglvel_z_calibbias(self, value):
+        self._set_iio_attr("anglvel_z", "calibbias", False, value)
+
+    @property
+    def anglvel_z_calibscale(self):
+        return self._get_iio_attr("anglvel_z", "calibscale", False)
```
Comment (tfcollins) [inline adi/adis16495.py:193]: Need docstring

File: adi/adis16495.py [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+    def accel_x_filter_low_pass_3db_frequency(self, value):
+        self._set_iio_attr("accel_x", "filter_low_pass_3db_frequency", False, value)
+
+    @property
+    def accel_x_calibbias(self):
+        return self._get_iio_attr("accel_x", "calibbias", False)
+
+    @accel_x_calibbias.setter
+    def accel_x_calibbias(self, value):
+        self._set_iio_attr("accel_x", "calibbias", False, value)
```
Comment (tfcollins) [inline adi/adis16495.py:221]: Need docstring

File: adi/adis16495.py [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+
+    @property
+    def accel_x_calibscale(self):
+        return self._get_iio_attr("accel_x", "calibscale", False)
```
Comment (tfcollins) [inline adi/adis16495.py:225]: Need docstring

File: adi/adis16495.py [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+        """ value returned in HZ """
+        return self._get_iio_attr("accel_y", "filter_low_pass_3db_frequency", False)
+
+    @accel_y_filter_low_pass_3db_frequency.setter
+    def accel_y_filter_low_pass_3db_frequency(self, value):
+        self._set_iio_attr("accel_y", "filter_low_pass_3db_frequency", False, value)
+
+    @property
+    def accel_y_calibbias(self):
+        return self._get_iio_attr("accel_y", "calibbias", False)
```
Comment (tfcollins) [inline adi/adis16495.py:249]: Need docstring

File: adi/adis16495.py [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+
+    @accel_y_calibbias.setter
+    def accel_y_calibbias(self, value):
+        self._set_iio_attr("accel_y", "calibbias", False, value)
+
+    @property
+    def accel_y_calibscale(self):
+        return self._get_iio_attr("accel_y", "calibscale", False)
```
Comment (tfcollins) [inline adi/adis16495.py:257]: Need docstring

File: adi/adis16495.py [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+    def accel_z_calibbias(self):
+        return self._get_iio_attr("accel_z", "calibbias", False)
+
+    @accel_z_calibbias.setter
+    def accel_z_calibbias(self, value):
+        self._set_iio_attr("accel_z", "calibbias", False, value)
+
+    @property
+    def accel_z_calibscale(self):
+        return self._get_iio_attr("accel_z", "calibscale", False)
```
Comment (tfcollins) [inline adi/adis16495.py:289]: Need docstring

**Comment by tfcollins** [comment]
Please add a doc entry in the devices folder

**Comment by nunojsa** [comment]
V2:

* Added missing docstrings and fixed some codacy complains;
* Added doc entry for new device.

**Comment by nunojsa** [comment]
V3:

* Fixed more codacy complains...

**Comment by nunojsa** [comment]
v4:

* More codacy stuff;
*  Added new class o index.rst and supported_parts.md

**Review by tfcollins — APPROVED** [review]

**Pull request merged**